workspace/src/control/src/control.cpp

```diff
@@ -0,0 +1,207 @@
+/******************************************************************************
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2022 University of Wisconsin - Madison
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+#include <iostream>
+#include <string>
+#include <chrono>
+#include <fstream>
+#include <algorithm>
+#include <functional>
+
+#include "control.h"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/qos.hpp"
+#include "ament_index_cpp/get_package_share_directory.hpp"
+
+control::control() : rclcpp::Node("control") {
+    this->startTime = this->get_clock()->now().nanoseconds() / 1e9;
+
+    /*
+        Update frequency of this node
+    */
+    this->nodeUpdateFrequency = 10.0;
+
+    /*
+        Default settings
+    */
+    this->mode = "PID";
```

Comment:
In C++, usually a trailing `_` or prefixing with `m_` replaces the need for `this`. I would recommend `m_` to stay consistent with Chrono.

Reply:
I thought this was more like a convention that programmers used. Is it actually a legit syntax?

Reply:
Idk why I was so uptight on my original review haha. I think maintaining coding practices between Chrono is good, of which uses `m_` prefix rather than `this->`. Just looking it up it looks like it's pretty debated as what is best practice, but I do think to defaulting to Chrono is best. Up for discussion tho.